### PR TITLE
[1LP][WIP]unified create in vm class

### PR DESCRIPTION
**Pull request by lkhomenk**
Purpose or Intent
=================
__Adding__ dicts with value for VM/Instance UI/REST provision and methods to use them

{{pytest: --long-running --use-provider complete cfme/tests/cloud/test_provisioning.py cfme/tests/infrastructure/test_provisioning.py cfme/tests/infrastructure/test_provisioning_rest.py cfme/tests/cloud/test_cloud_init_provisioning.py cfme/tests/cloud/test_tenant_quota.py}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 79851796e7c3dfd8c6fba9180de8b6c879addd52:

`cfme/common/vm.py`:
- :red_circle: Line 3:1: **F401** *'pytest' imported but unused*
- :red_circle: Line 6:1: **F401** *'fixtures.pytest_store.store' imported but unused*
- :red_circle: Line 8:1: **F401** *'novaclient.exceptions.OverLimit as OSOverLimit' imported but unused*
- :red_circle: Line 9:1: **F401** *'ovirtsdk.infrastructure.errors.RequestError as RHEVRequestError' imported but unused*
- :red_circle: Line 10:1: **F401** *'ssl.SSLError' imported but unused*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 41b4c99556606660c84bbf1c54985cfe4e95b348:

`cfme/common/vm.py`:
- :red_circle: Line 4:1: **F401** *'riggerlib.recursive_update' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/infrastructure/test_provisioning.py [hunk 1]
```diff
@@ -51,26 +51,15 @@ def test_infra_provision_from_template(appliance, setup_provider, provider, vm_n
...
-        }
-    }
+    inst_args = vm.vm_default_args
+    request.addfinalizer(lambda: vm.cleanup_on_provider())
 
-    do_vm_provisioning(appliance, template, provider, vm_name, provisioning_data, request,
-                       smtp_test, num_sec=900)
+    if auto:
+        inst_args['environment'] = {'automatic_placement': auto}
+    assert vm.create(check_existing=True, provisioning_data=inst_args, find_in_cfme=True), \
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_provisioning.py]: Required: use parentheses here
assert vm.create(check_existing=True, provisioning_data=inst_args, find_in_cfme=True), (
"VM {} wasn't created".format(vm.name))

File: cfme/tests/infrastructure/test_provisioning_rest.py [hunk 2]
```diff
@@ -82,9 +92,9 @@ def clean_vm(appliance, vm_name):
 
 
 @pytest.mark.rhv2
-# Here also available the ability to create multiple provision request, but used the save
-# href and method, so it doesn't make any sense actually
-def test_provision(request, appliance, provision_data):
+@pytest.mark.parametrize('auto', [True, False], ids=["Auto", "Manual"])
+def test_infra_provision_from_template_using_rest(request, appliance, vm_name, provider,
+                                              setup_provider, auto):
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_provisioning_rest.py]: Required: check indentation

**Comment by lkhomenk** [comment]
@mshriver @izapolsk  could you please check this PR?

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by lkhomenk** [comment]
replaced by #7335

**Pull request closed without merging**